crates/mod-rspamd/src/lib.rs

```diff
@@ -0,0 +1,340 @@
+use anyhow::Context;
+use config::{any_err, from_lua_value, get_or_create_sub_module};
+use data_loader::KeySource;
+use message::Message;
+use mlua::prelude::LuaUserData;
+use mlua::{Lua, LuaSerdeExt, UserDataMethods, Value};
+use rspamd_client::config::{Config, EnvelopeData};
+use rspamd_client::protocol::RspamdScanReply;
+use rspamd_client::scan_async;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// Configuration for Rspamd client (Lua-friendly wrapper)
+#[derive(Deserialize, Debug, Clone)]
+struct RspamdClientConfig {
+    /// Base URL for Rspamd server (e.g., "http://localhost:11333")
+    /// Must be a valid URL
+    base_url: url::Url,
+
+    /// Optional password for Rspamd authentication
+    #[serde(default)]
+    password: Option<KeySource>,
+
+    /// Optional timeout in seconds (default: 30.0)
+    #[serde(default, with = "duration_serde")]
+    timeout: Option<std::time::Duration>,
+
+    /// Number of retries for requests (default: 1)
+    #[serde(default)]
+    retries: Option<u32>,
+
+    /// Enable ZSTD compression (default: true)
+    /// Automatically compresses message bodies for faster transmission
+    #[serde(default = "default_true")]
+    zstd: bool,
+
+    /// Optional HTTPCrypt encryption key (base32 format)
+    /// When set, enables encrypted communication with Rspamd
+    /// Generate with: rspamadm keypair
+    /// Supports KeySource for secure storage
+    #[serde(default)]
+    encryption_key: Option<KeySource>,
+
+    /// Optional proxy URL (e.g., "http://proxy.example.com:8080")
+    #[serde(default)]
+    proxy_url: Option<String>,
+
+    /// Optional proxy username
+    /// Supports KeySource for secure storage
+    #[serde(default)]
+    proxy_username: Option<KeySource>,
+
+    /// Optional proxy password
+    /// Supports KeySource for secure storage
+    #[serde(default)]
+    proxy_password: Option<KeySource>,
+
+    /// If true, add X-Spam-* headers to the message (default: true)
+    /// Headers added: X-Spam-Flag, X-Spam-Score, X-Spam-Action, X-Spam-Symbols
+    #[serde(default = "default_true")]
+    add_headers: bool,
+
+    /// If true, prefix the Subject header with "***SPAM*** " when action is "rewrite subject" (default: false)
+    /// Note: This only applies when Rspamd returns action "rewrite subject"
+    #[serde(default)]
+    prefix_subject: bool,
+
+    /// If true, reject messages when Rspamd action is "reject" (default: false)
+    /// If false, messages with action "reject" will still be delivered with spam headers
+    /// Note: Action "soft reject" always results in 451 temporary failure (greylisting)
```

Comment:
is there a straight forward way for a user to prevent rspamd returning a soft reject?  I can see this sort of unilateral policy decision being something that one of our users will ask for a way to override so that they can handle it for themselves in a possible weird/inventive way!